src/main/java/com/redhat/devtools/lsp4ij/client/features/LSPOnTypeFormattingFeature.java

```diff
@@ -0,0 +1,96 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at https://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.client.features;
+
+import com.intellij.lang.LanguageFormatting;
+import com.intellij.psi.PsiFile;
+import com.redhat.devtools.lsp4ij.ServerStatus;
+import com.redhat.devtools.lsp4ij.server.capabilities.OnTypeFormattingCapabilityRegistry;
+import org.eclipse.lsp4j.ServerCapabilities;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * LSP onTypeFormatting feature.
+ */
+@ApiStatus.Experimental
+public class LSPOnTypeFormattingFeature extends AbstractLSPDocumentFeature {
+
+    private OnTypeFormattingCapabilityRegistry onTypeFormattingCapabilityRegistry;
+
+    @Override
+    public boolean isEnabled(@NotNull PsiFile file) {
+        // This feature should only be enabled if the server is already started
+        if (getServerStatus() != ServerStatus.started) {
```

Comment:
Please add a comment why you do that (for permanence raison)

Reply:
Done.